src/overlays/actors/ovl_Bg_Ydan_Hasi/z_bg_ydan_hasi.c

```diff
@@ -21,23 +27,177 @@ const ActorInit Bg_Ydan_Hasi_InitVars = {
     (ActorFunc)BgYdanHasi_Update,
     (ActorFunc)BgYdanHasi_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/BgYdanHasi_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/BgYdanHasi_Destroy.s")
+static InitChainEntry sInitChain[] = {
+
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+extern Gfx* D_808BEC24[] = { 0x06007508, 0x06005DE0, 0x06005018 };
+extern UNK_TYPE D_06005780;
+extern UNK_TYPE D_06007798;
+extern Gfx* D_06005DE0;
+
+void BgYdanHasi_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s16 pad1;
```

Comment:
Avoid non-s32 pads unless necessary.
```suggestion
    s32 pad1;
```